solarwinds_apm/apm_config.py

```diff
@@ -93,23 +89,8 @@ def __init__(
                 "enabled"
             ),
             "collector": "",  # the collector address in host:port format.
-            "reporter": "",  # the reporter mode, either 'udp' or 'ssl'.
-            "log_type": apm_logging.ApmLoggingType.default_type(),
             "debug_level": apm_logging.ApmLoggingLevel.default_level(),
-            "log_filepath": "",
             "service_key": "",
-            "hostname_alias": "",
-            "trustedpath": "",
-            "events_flush_interval": -1,
-            "max_request_size_bytes": -1,
-            "ec2_metadata_timeout": 1000,
```

Comment:
Could we also clean up the helper function `_set_config_value`? It no longer needs the conditions for setting `ec2_metadata_timeout`, `proxy`, etc.

Reply:
We still need that for testing other fields in the apm-config like `service_key`, `transaction_filters`. 
However, it can be improved when we replace the entire `__config` data structure.

Reply:
We don't need to completely remove `_set_config_value`. It could be updated so that instead of starting with

```
        try:
            if keys == ["ec2_metadata_timeout"]:
```

it could have the conditions for `ec2...`, `proxy`, `tracing_mode` removed, then start with

```
        try:
            if keys == ["trigger_trace"]:
```

If it's easier though then we can leave it alone and focus later on the overhaul of `__config`.

Reply:
Either way, that would need a new ticket. Would that be in our current epic, or later?

Reply:
Use this ticket to housekeep old/dead code https://swicloud.atlassian.net/browse/NH-104999